modules/fedimint-ln-client/src/lib.rs

```diff
@@ -617,15 +652,17 @@ impl LightningClientModule {
         Ok((client_output, contract_id))
     }
 
+    /// Returns a bool indicating if it was an external receive
     async fn await_receive_success(
         &self,
         operation_id: OperationId,
-    ) -> Result<(), LightningReceiveError> {
+    ) -> Result<bool, LightningReceiveError> {
         let mut stream = self.notifier.subscribe(operation_id).await;
         loop {
             match stream.next().await {
                 Some(LightningClientStateMachines::Receive(state)) => match state.state {
-                    LightningReceiveStates::Funded(_) => return Ok(()),
+                    LightningReceiveStates::Funded(_) => return Ok(false),
+                    LightningReceiveStates::Success(_) => return Ok(true),
```

Comment:
I believe `LightningReceiveStates::Success` can happen when the payment is an internal receive as well, but when its an internal receive `Funded` should happen first. I think there might be a small race condition where if you call `await_receive_success` after `Funded` has been emitted for an internal payment, this function will return true even if its an internal payment.

Reply:
changed to return `outpoints.is_empty()` this should only be true for external receives